tests/pastefromoffice.js

```diff
@@ -39,80 +40,63 @@ describe( 'PasteFromOffice', () => {
-			} );
+		it( 'should process data with nested microsoft header', () => {
+			checkDataProcessing( '<html><head><meta name="Generator"  content=Microsoft Word 15></head></html>', true );
 		} );
 
-		describe( 'already processed data', () => {
-			[
-				{
-					// eslint-disable-next-line max-len
-					'text/html': '<meta charset="utf-8"><b id="docs-internal-guid-30db46f5-7fff-15a1-e17c-1234567890ab" style="font-weight:normal;"><p dir="ltr">Hello world</p></b>'
-				},
-				{
-					'text/html': '<meta name=Generator content="Microsoft Word 15"><p class="MsoNormal">Hello world<o:p></o:p></p>'
-				}
-			].forEach( ( inputData, index ) => {
-				it( `should not modify already processed data: #${ index }`, () => {
-					const data = {
-						content: htmlDataProcessor.toView( inputData[ 'text/html' ] ),
-						dataTransfer: createDataTransfer( inputData ),
-						isTransformedWithPasteFromOffice: true
-					};
-
-					const getData = sinon.spy( data.dataTransfer, 'getData' );
-
-					clipboard.fire( 'inputTransformation', data );
-
-					// Data object should not be processed
-					sinon.assert.notCalled( getData );
-				} );
-			} );
+		it( 'should process data from google docs', () => {
+			checkDataProcessing( '<p id="docs-internal-guid-12345678-1234-1234-1234-1234567890ab"></p>', true );
+		} );
+
+		it( 'should not process data with regular html', () => {
+			checkDataProcessing( '<p>Hello world</p>', false );
+		} );
+
+		it( 'should not process data with similar headers to MS Word', () => {
+			checkDataProcessing( '<meta name=Generator content="Other">', false );
+		} );
+
+		it( 'should not process again ms word data containing a flag', () => {
+			checkDataProcessing( '<meta name=Generator content="Microsoft Word 15"><p class="MsoNormal">Hello world<o:p></o:p></p>',
+				false, true );
+		} );
+
+		it( 'should not process again google docs data containing a flag', () => {
+			checkDataProcessing( '<meta charset="utf-8"><b id="docs-internal-guid-30db46f5-7fff-15a1-e17c-1234567890ab"' +
+				'style="font-weight:normal;"><p dir="ltr">Hello world</p></b>', false, true );
 		} );
 	} );
+
+	// @param {String} inputString html to be processed by paste from office
+	// @param {Boolean} shouldBeProcessed determines if data should be marked as processed with isTransformedWithPasteFromOffice flag
+	// @param {Boolean} [isAlreadyProcessed=false] apply flag before paste from office plugin will transform the data object
+	function checkDataProcessing( inputString, shouldBeProcessed, isAlreadyProcessed = false ) {
+		// const htmlDataProcessor = new HtmlDataProcessor();
+		const data = {
+			content: htmlDataProcessor.toView( inputString ),
+			dataTransfer: createDataTransfer( { 'text/html': inputString } )
+		};
+		const getData = sinon.spy( data.dataTransfer, 'getData' );
+
+		if ( isAlreadyProcessed ) {
+			data.isTransformedWithPasteFromOffice = true;
+		}
+
+		clipboard.fire( 'inputTransformation', data );
+
+		if ( shouldBeProcessed ) {
```

Comment:
Those two methods of testing could be preserved - now the test is a bit too mangled ;) Also two booleans in method parameters are too much.